internal/cmd/app.go

```diff
@@ -84,21 +88,36 @@ func newApp(newRegistry internal.NewRegistry) *cli.App {
 			if err != nil {
 				return err
 			}
-			layerPattern, err = validateLayerPatternFlag(c.String(flagLayerPattern))
-			return err
+			createdByPattern, err = validateCreatedByPatternFlag(c.String(flagCreatedByPattern))
+			if err != nil {
+				return err
+			}
+			if c.Bool(flagExtract) {
+				directory, err = validateDirectoryFlag(c.String(flagDirectory))
+				if err != nil {
+					return err
+				}
+				stripComponents, err = validateStripComponentsFlag(c.Int(flagStripComponents))
+				if err != nil {
+					return err
+				}
+			}
+			return nil
 		},
 		Action: func(c *cli.Context) error {
 			car := carutil.New(
 				newRegistry(c.Context, domain, path),
 				c.App.Writer,
-				layerPattern,
+				createdByPattern,
 				c.Args().Slice(),
 				c.Bool(flagFastRead),
 				c.Bool(flagVerbose),
 				c.Bool(flagVeryVerbose),
 			)
 			if c.Bool(flagList) {
 				return car.List(c.Context, tag, platform)
+			} else if c.Bool(flagExtract) {
```

Comment:
not sure how `tar` behaves, but is this aligned with it? I mean `list` is prioritized over `extract` rather than emitting errors?

Reply:
when extract and list are both set

Reply:
I'll break on incompatible flags. good idea!